### ISSUE TITLE
The "check" task does not detect that the "verifyPaparazziDebug" task has failed.

### ISSUE THREAD

**Issue by CharlesMoreira1** [cashapp/paparazzi#1102]
**Description**

I was conducting some tests with Paparazzi by running the "verifyPaparazziDebug" task, and I noticed that it works well, detecting when it fails. However, when I tried running the "check" task, it doesn't detect the failure. Is this expected behavior, or is it possibly a bug?

I needed to run it as a "check" since we don't have access to the CI here, and we needed to detect any image discrepancies.

**Additional information:**
- Paparazzi Version: 1.3.1
- OS: macOs Ventura 13.3.1
- Compile SDK: 34
- Gradle Version: 8.1.1
- Android Gradle Plugin Version: 8.2

**Comment by TWiStErRob** [comment]
If you look at how the verify task is registered (see where the variable is used):

https://github.com/cashapp/paparazzi/blob/33963c034abdebf02ccd007ff0c4af4ed6226c4d/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L210

you can see that it's not hooked to the [`check`](https://docs.gradle.org/current/userguide/base_plugin.html#sec:base_tasks) standard [lifecycle task](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:lifecycle_tasks).

If you have no access to your CI (i.e. can't modify what tasks it executes), you can surely edit your build files. In every paparazzi module you can do something like this:
```gradle.kts
androidComponents {
    onVariants { variant ->
        tasks.named("check").configure { dependsOn("verifyPaparazzi${variant.name.capitalize(Locale.ROOT)}") }
    }
}
```
Depending on your requirements, you might want to put it under an `if (isCI)` condition so that when developers locally run `gradlew build` they don't run these "integration tests" (`build` depends on `check`).

**Comment by CharlesMoreira1** [comment]
It worked perfectly. I didn't know I could do that.

But I thought it already ran on check, because when I did `./gradlew tasks`, it was listed under

```
Verification tasks
------------------
```

Anyway, you were able to help me a lot. Thank you very much!

**Comment by TWiStErRob** [comment]
No, that's simply [group](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#setGroup-java.lang.String-):

https://github.com/cashapp/paparazzi/blob/33963c034abdebf02ccd007ff0c4af4ed6226c4d/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L211

You can assign it anything, it's for humans only.

**Comment by CharlesMoreira1** [comment]
> No, that's simply [group](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#setGroup-java.lang.String-):
> 
> https://github.com/cashapp/paparazzi/blob/33963c034abdebf02ccd007ff0c4af4ed6226c4d/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L211
> 
> You can assign it anything, it's for humans only.

Interesting. I'll delve deeper into this. Thank you! <3

**Comment by CharlesMoreira1** [comment]
> No, that's simply [group](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#setGroup-java.lang.String-):
> 
> https://github.com/cashapp/paparazzi/blob/33963c034abdebf02ccd007ff0c4af4ed6226c4d/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L211
> 
> You can assign it anything, it's for humans only.

Interesting. I'll delve deeper into this. Thank you! <3

**Comment by Goooler** [comment]
Created a new FR for this, see #2232.
